Sources/UI/Device/Device.swift

```diff
@@ -0,0 +1,405 @@
+//
+//  Device.swift
+//  RoverUI
+//
+//  Created by Andrew Clunis on 2018-11-20.
+//  Copyright © 2018 Rover Labs Inc. All rights reserved.
+//
+
+import Foundation
+import os
+import UIKit
+import UserNotifications
+
+class Device {
+    
+    let userDefaults: UserDefaults
+    let jsonDecoder: JSONDecoder
+    let jsonEncoder: JSONEncoder
+    
+    let adSupportInfoProvider: AdSupportInfoProvider?
+    let bluetoothInfoProvider: BluetoothInfoProvider?
+    let telephonyInfoProvider: TelephonyInfoProvider?
+    let locationInfoProvider: LocationInfoProvider?
+    
+    let reachability = Reachability(hostname: "google.com")!
+    let userNotificationCenter = UNUserNotificationCenter.current()
+    
+    public init(
+        userDefaults: UserDefaults,
+        jsonDecoder: JSONDecoder,
+        jsonEncoder: JSONEncoder,
+        adSupportInfoProvider: AdSupportInfoProvider?,
+        bluetoothInfoProvider: BluetoothInfoProvider?,
+        telephonyInfoProvider: TelephonyInfoProvider?,
+        locationInfoProvider: LocationInfoProvider?
+    ) {
+        self.userDefaults = userDefaults
+        self.jsonDecoder = jsonDecoder
+        self.jsonEncoder = jsonEncoder
+        self.adSupportInfoProvider = adSupportInfoProvider
+        self.bluetoothInfoProvider = bluetoothInfoProvider
+        self.telephonyInfoProvider = telephonyInfoProvider
+        self.locationInfoProvider = locationInfoProvider
+    }
+    
+    public var snapshot: DeviceSnapshot {
+        return DeviceSnapshot(
+            advertisingIdentifier: advertisingIdentifier,
```

Comment:
Can you refer to all of these with `self`? I used to omit the `self` keyword but have found it adds clarity and reduces bugs. It is immediately obvious to developers "ah this property is defined as an instance variable on the class" instead of "where did this `advertisingIdentifer` variable come from? Is it a function variable or a instance property of the class?".